File: Application/Services/Operations/Finances/Dtos/FinancingsLoansExpenses/FinancingAndLoanExpenseInstallmentDto.cs
using System;
using System.Collections.Generic;
using Application.Services.Operations.Authentication.Dtos;
using Application.Services.Operations.Finances.Dtos.Bank;
using Application.Services.Operations.Finances.Dtos.InheritanceDto;
using Application.Services.Operations.Finances.Dtos.PixExpenses;
using Application.Services.Operations.Main.Companies.Dtos;
using Application.Services.Shared.Dtos;
using Domain.Entities.Finances.Inheritance;

namespace Application.Services.Operations.Finances.Dtos.FinancingsLoansExpenses
{
    public class FinancingAndLoanExpenseInstallmentDto:RootBaseDto
    {
        public int? BankAccountId { get; set; }
        public BankAccountDto BankAccount { get; set; }
        public int? CardId { get; set; }
        public CardDto Card { get; set; }
        public int? PixId { get; set; }
        public PixDto Pix { get; set; }
        public PixExpenseDto PixExpense {get; set;}
        public decimal Interest { get; set; }
        public DateTime Expires { get; set; }
        public DateTime WasPaid { get; set; }
        public string OthersPaymentMethods { get; set; }
        public string Document { get; set; }
        public decimal PriceWasPaidInstallment { get; set; }
        public string CurrentInstallment { get; set; }
        public int FinancingAndLoanExpenseId { get; set; }
        public FinancingAndLoanExpenseDto FinancingAndLoanExpense { get; set; }
    }
}

File: Views/src/components/financial/components/financings-loans-expenses/dto/financing-and-loan-expense-installment-dto.ts
import { MyUser } from "src/components/authentication/dto/my-user";
import { CardDto } from "../../bank-account-cards/dto/card-dto";
import { BaseExpenseDto } from "../../common-components/dto/base-expense-dto";
import { BankAccountDto } from "../../bank-account-cards/dto/bank-account-dto";
import { PixDto } from "../../bank-account-cards/dto/pix-dto";
import { CompanyDto } from "src/shared/entities-dtos/company-dto";
import { FinancingsLoansExpensesDto } from "./financings-loans-expenses-dto";
import { RootBase } from "src/shared/entities-dtos/root-base";

export class FinancingAndLoanExpenseInstallmentDto extends RootBase {
  bankAccountId: number;
  bankAccount: BankAccountDto;
  cardId: number;
  card: CardDto;
  pixId: number;
  pix: PixDto;
  interest: number;
  priceWasPaidInstallment: number;
  expires: Date;
  wasPaid: Date;
  othersPaymentMethods: string;
  document: string;
  currentInstallment: string;
  financingAndLoanExpense: FinancingsLoansExpensesDto;
  financingAndLoanExpenseId: number;
}
